qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -0,0 +1,526 @@
+        return measurement_data
+
+    def solve(self, schedule: Union[ScheduleBlock, Schedule], qubits: Tuple[int]) -> np.ndarray:
+        """Solves for qubit dynamics under the action of a pulse instruction
+
+        Args:
+            schedule: Pulse signal
+            qubits: (remove after multi-qubit gates is implemented)
+
+        Returns:
+            Time-evolution unitary operator
+        """
+        if len(qubits) > 1:
+            QiskitError("Multi qubit gates are not yet implemented.")
+
+        if isinstance(schedule, ScheduleBlock):
+            schedule = block_to_schedule(schedule)
+
+        signal = self.converter.get_signals(schedule)
+        time_f = schedule.duration * self.dt
+        unitary = self.solver.solve(
+            t_span=[0.0, time_f],
+            y0=self.y_0,
+            t_eval=[time_f],
+            signals=signal,
+            method=self.solver_method,
+        ).y[0]
+
+        return unitary
+
+    def run(self, run_input: Union[QuantumCircuit, List[QuantumCircuit]], **run_options) -> FakeJob:
+        """run method takes circuits as input and returns FakeJob with IQ data or counts.
+
+        Args:
+            run_input: Circuits to run
+
+        Returns:
+            FakeJob with simulation data.
+        """
+        self.options.update_options(**run_options)
+        shots = self.options.get("shots", self._options.shots)
+        meas_level = self.options.get("meas_level", self._options.meas_level)
+        meas_return = self.options.get("meas_return", self._options.meas_return)
+
+        result = {
+            "backend_name": f"{self.__class__.__name__}",
+            "backend_version": self.backend_version,
+            "qobj_id": 0,
+            "job_id": 0,
+            "success": True,
+            "results": [],
+        }
+
+        if isinstance(run_input, QuantumCircuit):
+            run_input = [run_input]
+
+        for circuit in run_input:
+            unitaries = {}
+
+            # 1. Parse the calibrations and simulate any new schedule. Add U to the unitaries.
+            for name, schedule in circuit.calibrations.items():
+                for (qubits, params), schedule_block in schedule.items():
+                    schedule_key = hash(repr(schedule))
+
+                    # Simulate the schedule if not in the cache.
+                    if schedule_key not in self._schedule_cache:
+                        self._schedule_cache[schedule_key] = self.solve(schedule_block, qubits)
+
+                    unitaries[(name, qubits, params)] = self._schedule_cache[schedule_key]
+
+            # 2. Copy over any remaining instructions to the dict of unitaries.
+            for key, unitary in self.default_pulse_unitaries.items():
+                if key not in unitaries:
+                    unitaries[key] = unitary
+
+            # 3. Multiply the unitaries of the circuit instructions onto the ground state.
+            state_t = np.array([1.0] + [0.0] * (self.model_dim - 1))
```

Comment:
A bug that is missed previously, model_dim must remain `3` for density matrix reshaping and IQ shot generation. We need the value `9` only for the `y0` and `gs` definitions. So maybe it's better to have them defined in one place in the init. Something like:
```
if self.static_dissipators is None:
    self.y_0 = np.eye(self.model_dim)
    self.gs = np.array([1.0] + [0.0] * (self.model_dim - 1))
else:
    self.y_0 = np.eye(self.model_dim**2)
    self.gs = np.array([1.0] + [0.0] * (self.model_dim**2 - 1))
```

Reply:
Alternatively we could do:
```
self.model_dim = self.physical_levels = self.solver.model.dim
if self.static_dissipators is not None:
    self.model_dim = self.model_dim**2
```
and the use `self.physical_levels` for IQ and DM reshaping